imaging/ShrinkPixelator.cpp

```diff
@@ -1,26 +1,141 @@
-#include "ShrinkPixelator.h"
-
-namespace imaging
-{
-    ShrinkPixelator::ShrinkPixelator(const std::string& in_path) : Pixelator{ in_path } {}
-
-    bool ShrinkPixelator::to_pixels(int rows, int columns)
-    {
-        ImgData buffer;
-        int rows_backup = pictureBuffer.rows;
-        int columns_backup = pictureBuffer.cols;
-        try
-        {
-            cv::resize(pictureBuffer, buffer, cv::Size(columns, rows));
-            cv::swap(pictureBuffer, buffer);
-            binarize();
-            cv::swap(pictureBuffer, buffer);
-            cv::resize(buffer, pictureBuffer, cv::Size(columns_backup, rows_backup), 0.0, 0.0, cv::InterpolationFlags::INTER_NEAREST);
-            return true;
-        }
-        catch (...)
-        {
-            return false;
-        }
-    }
-}
\ No newline at end of file
+#include "ShrinkPixelator.h"
+#include <optional>
+
+namespace
+{
+  unsigned greatest_common_divisor(unsigned a, unsigned b);
+}
+
+namespace imaging
+{
+  ShrinkPixelator::ShrinkPixelator(const std::string& in_path) : Pixelator{ in_path } {}
+
+  bool ShrinkPixelator::to_pixels(int in_width, int in_height, int gauge_stitches, int gauge_rows, const one_bit::CropRegion in_cropRegion)
+  {
+    ImgData buffer;
+    const double aspect_ratio_epsilon = 0.01;
+    int rows_backup = pictureBuffer.rows;
+    int columns_backup = pictureBuffer.cols;
+    double aspect_ratio_old = (rows_backup * 1.) / columns_backup; 
+    unsigned stixelGcd{ greatest_common_divisor(gauge_rows, gauge_stitches) };
```

Comment:
This got fixed for the UI version. Either abandon the shell version or fix this.